File: ExampleMvvm/librarycommon/src/main/java/com/degde/library/library/login/viewmodels/MainViewModel.java
package com.degde.library.library.login.viewmodels;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.degde.library.library.login.model.ModelLoginResponse;
import com.degde.library.library.login.networking.LoginRepository;


public class MainViewModel extends ViewModel {
    public static MutableLiveData<ModelLoginResponse> mutableLiveData;
    private LoginRepository loginRepository;


    public void initRequest(){
        if (mutableLiveData != null){
            return;
        }
        loginRepository = LoginRepository.getInstance();
        mutableLiveData = loginRepository.serviceRequest();

    }
    public MutableLiveData<ModelLoginResponse> getNewsRepository() {
        return mutableLiveData;
    }


}

File: ExampleMvvm/corekotlin/src/main/java/com/core/mvvm/login/viewmodel/MainViewModel.kt
package com.core.mvvm.login.viewmodel

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.core.mvvm.login.model.ModelLoginResponse
import com.core.mvvm.login.networking.LoginRepository


class MainViewModel : ViewModel() {
    private var loginRepository: LoginRepository? = null
    companion object {
        var mutableLiveData: MutableLiveData<ModelLoginResponse>? = null
    }

    fun initRequest() {
        if (mutableLiveData != null) {
            return
        }
        loginRepository = LoginRepository.getInstance()
        mutableLiveData = loginRepository!!.serviceRequest()
    }



    fun newsRepository() :  MutableLiveData<ModelLoginResponse>? {
        return mutableLiveData
    }


}



